api/v1alpha1/oidc_types.go

```diff
@@ -48,8 +49,50 @@ type OIDC struct {
 	RedirectURL *string `json:"redirectURL,omitempty"`
 
 	// The path to log a user out, clearing their credential cookies.
+	//
 	// If not specified, uses a default logout path "/logout"
 	LogoutPath *string `json:"logoutPath,omitempty"`
+
+	// ForwardBearerToken indicates whether the Envoy should forward the access
+	// token as a bearer token in the "Authorization" header to the backend.
+	//
+	// If not specified, defaults to false.
+	// +optional
+	// +notImplementedHide
+	ForwardBearerToken *bool `json:"forwardBearerToken,omitempty"`
+
+	// The default lifetime of the ID token and access token.
+	// Please note that Envoy will always use the expiry time from the response
+	// of the authorization server if it is provided. This field is only used when
+	// the expiry time is not provided by the authorization.
+	//
+	// If not specified, defaults to 0. In this case, the "expires_in" field in
+	// the authorization response must be set by the authorization server, or the
+	// OAuth flow will fail.
+	//
+	// +optional
+	// +notImplementedHide
+	DefaultTokenExpireTime *metav1.Duration `json:"defaultTokenExpireTime,omitempty"`
```

Comment:
should this be `DefaultIdTokenExpireTime` ?

Reply:
The Envoy doc is somewhat vague in[ its description](https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/filters/http/oauth2/v3/oauth.proto#envoy-v3-api-msg-extensions-filters-http-oauth2-v3-oauth2config) of `DefaultTokenExpireTime`.

However, after digging into the code, it's clear that this applies to both the ID token and Access Token.

https://github.com/envoyproxy/envoy/blob/a112766683a6b82d20843c70b8aa4d7eda29048e/source/extensions/filters/http/oauth2/oauth_client.cc#L183

Reply:
Or `DefaultIdAndAccessTokenExpireTime`, that's quite lengthy :-)